bee-api/bee-rest-api/src/endpoints/config.rs

```diff
@@ -1,84 +1,60 @@
-pub(crate) const ROUTE_TIPS: &str = "/api/v1/tips";
-pub(crate) const ROUTE_RECEIPTS: &str = "/api/v1/receipts";
-pub(crate) const ROUTE_RECEIPTS_AT: &str = "/api/v1/receipts/:milestoneIndex";
-pub(crate) const ROUTE_TREASURY: &str = "/api/v1/treasury";
-pub(crate) const ROUTE_TRANSACTION_INCLUDED_MESSAGE: &str = "/api/v1/transactions/:transactionId/included-message";
-pub(crate) const ROUTE_WHITE_FLAG: &str = "/api/plugins/debug/whiteflag";
-
-/// the routes that are available for public use
-pub(crate) const DEFAULT_PUBLIC_ROUTES: [&str; 21] = [
-    ROUTE_BALANCE_BECH32,
-    ROUTE_BALANCE_ED25519,
-    ROUTE_HEALTH,
-    ROUTE_INFO,
-    ROUTE_MESSAGE,
-    ROUTE_MESSAGE_CHILDREN,
-    ROUTE_MESSAGE_METADATA,
-    ROUTE_MESSAGE_RAW,
-    ROUTE_MESSAGES_FIND,
-    ROUTE_MILESTONE,
-    ROUTE_MILESTONE_UTXO_CHANGES,
-    ROUTE_OUTPUT,
-    ROUTE_OUTPUTS_BECH32,
-    ROUTE_OUTPUTS_ED25519,
-    ROUTE_SUBMIT_MESSAGE,
-    ROUTE_SUBMIT_MESSAGE_RAW,
-    ROUTE_TIPS,
-    ROUTE_RECEIPTS,
-    ROUTE_RECEIPTS_AT,
-    ROUTE_TREASURY,
-    ROUTE_TRANSACTION_INCLUDED_MESSAGE,
-];
-pub(crate) const DEFAULT_ALLOWED_IPS: [IpAddr; 2] = [
-    IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)),
-    IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1)),
+/// Default JWT salt for REST API.
+pub(crate) const DEFAULT_JWT_SALT: &str = "Bee";
+/// Default routes that are available for public use and don't need JWT authentication.
+pub(crate) const DEFAULT_PUBLIC_ROUTES: [&str; 11] = [
+    "/health",
+    "/mqtt",
+    "/api/v1/info",
+    "/api/v1/tips",
+    "/api/v1/messages*",
+    "/api/v1/transactions*",
+    "/api/v1/milestones*",
+    "/api/v1/outputs*",
+    "/api/v1/addresses*",
+    "/api/v1/treasury",
+    "/api/v1/receipts*",
 ];
+/// Default routes that are protected and need JWT authentication.
+pub(crate) const DEFAULT_PROTECTED_ROUTES: [&str; 2] = ["/api/v1/*", "/api/plugins/*"];
+/// Enables the proof-of-work feature on the node per default.
 pub(crate) const DEFAULT_FEATURE_PROOF_OF_WORK: bool = true;
-pub(crate) const DEFAULT_WHITE_FLAG_SOLIDIFICATION_TIMEOUT: u64 = 2;
+/// Default value for the white flag solidification timeout.
+pub(crate) const DEFAULT_WHITE_FLAG_SOLIDIFICATION_TIMEOUT: Duration = Duration::from_secs(2);
 
 /// REST API configuration builder.
 #[derive(Default, Deserialize, PartialEq)]
 #[must_use]
 pub struct RestApiConfigBuilder {
+    /// REST API binding address.
     #[serde(alias = "bindAddress")]
     bind_address: Option<Multiaddr>,
+    /// JWT salt for REST API.
+    #[serde(alias = "jwtSalt")]
+    jwt_salt: Option<String>,
+    /// Routes that are available for public use and don't need JWT authentication.
     #[serde(alias = "publicRoutes")]
     public_routes: Option<Vec<String>>,
-    #[serde(alias = "allowedIps")]
-    allowed_ips: Option<Vec<IpAddr>>,
+    /// Routes that are protected and need JWT authentication.
+    #[serde(alias = "protectedRoutes")]
+    protected_routes: Option<Vec<String>>,
```

Comment:
Classical question: does the difference between `Some(vec![])` and `None` actually matter?

Reply:
If not, I'd prefer `Vec<String>`.

Reply:
It does matter, `None` notifies that it wasn't set at all, triggering the default value for it.

Reply:
So let's keep it